sources/api/netdog/src/main.rs

```diff
@@ -10,13 +10,23 @@ It contains two subcommands meant for use as settings generators:
 * `generate-hostname`: returns the node's hostname in JSON format. If the lookup is unsuccessful, the IP of the node is used.
 
 The subcommand `set-hostname` sets the hostname for the system.
+
+The subcommand `generate-net-config` generates the network interface configuration for the host.
+If a `net.toml` exists, it is used to generate the configuration, otherwise a default configuration
+is written for interface `eth0` (AWS/VMware variants) or `eno1` (other variants).
 */
 
 #![deny(rust_2018_idioms)]
 
 #[macro_use]
 extern crate serde_plain;
 
+mod net_config;
+use net_config::NetConfig;
+
+mod wicked;
+use wicked::WickedInterface;
+
```

Comment:
nit: is there a reason to keep these use statements separate from all the others? rust fmt will just sort the lists of mods and uses if they are aggregated.